src/coreComponents/codingUtilities/Table.hpp

```diff
@@ -0,0 +1,251 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 TotalEnergies
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file Table.hpp
+ */
+
+#ifndef GEOS_COMMON_TABLE_HPP
+#define GEOS_COMMON_TABLE_HPP
+
+#include "common/DataTypes.hpp"
+
+namespace geos
+{
+
+class Table
+{
+public:
+
+  enum Alignment { right, left, middle };
+
+  enum MarginType {border, column};
+
+  enum MarginValue { tiny, small, medium, large};
+
+  /**
+   * @brief Struct for a column margin and border margin.
+   * @param tiny 0 margin for both;
+   * @param small 1 margin from |
+   * @param medium 2 margin from |
+   * @param large 3 margin from |
+   *
+   * @param marginValue current margin value
+   */
+  struct Margin
+  {
+    integer tiny;
+    integer small;
+    integer medium;
+    integer large;
+
+    integer marginValue;
+
+    void setWorkingValue( integer const & value )
+    {
+      marginValue = value;
+    }
+  };
+
+  /**
+   * @brief Struct for a ColumnParam.
+   * @param headerName A vector containing all string for a header name
```

Comment:
You can use `@var` to document a struct, but as you can see you can forget to update a variable name. Personally, I'd have prefer this syntax :
```
  /**
   * @brief Struct for a ColumnParam.
   */
  struct ColumnParam
  {
    /// A vector containing all header name strings
    std::vector< string > headerName;
```